src/posix/platform/system.cpp

```diff
@@ -94,6 +95,11 @@ otInstance *otSysInit(otPlatformConfig *aPlatformConfig)
     platformBackboneInit(instance, aPlatformConfig->mBackboneInterfaceName);
 #endif
 
+#if OPENTHREAD_CONFIG_BORDER_ROUTING_ENABLE
+    // Reuse the backbone interface name.
+    platformInfraIfInit(instance, aPlatformConfig->mBackboneInterfaceName);
```

Comment:
Just reuse `platformBackboneInit`? Maybe rename `platformBackboneInit` to `platformInfraIfInit`.

Reply:
For now, I think it is good to have them decoupled so that they will not affect each other. For future refactoring of the multicast feature, sure we can refactor to use the single backbone/infrastructure interface representation. But for now, I am not very clear about best representation that can satisfy both border routing and multicasting.

And looks to me there is only a small portion of the initialization code that can be reused (create a ICMPv6 secoket).

I'd like to keep this PR simple and clean so that it includes only necessary changes to bring up the border routing feature.